### ISSUE TITLE
Position of the car

### ISSUE THREAD

**Issue by Davinho10** [eleurent/highway-env#424]
Are the  x and y values measured in the kinematics observations the middle point of the cars? 

![image](https://user-images.githubusercontent.com/82339384/222212487-74936633-aeee-49ef-ae71-cd987f4189c9.png)

**Comment by eleurent** [comment]
yes, absolutely 👍

**Comment by Davinho10** [comment]
Just to clarify , when I have the x and y coordinates dependent on the ego vehicle. Is it the difference from the middle point or is it the difference from the front of the ego vehicle to the back of for example vehicle 2?

**Comment by eleurent** [comment]
Vehicle positions are always measured at their centrepoint, so it is the difference between the centres of the two vehicles.

**Comment by Davinho10** [comment]
![image](https://user-images.githubusercontent.com/82339384/233095400-c95bcfc3-2e25-4a93-a0b6-8a05b5031175.png)
I have a question regarding the Crash and the related Distance between the cars.
I get the Crash Variable from the step function :
obs, reward, done, info = env.step(int(action))
Crash = info["crashed"]
Also I save the x and y coordinates which I get from: dest_state = obs.tolist()
Shouldnt it be a crash when the y coordinates are less then 2 from each other because of the width of the cars from the middle point(1m) and the x coordinates smaller then 5 because of each of the cars length(2.5m)?

Because in this case the y distance would be less then 2 but the x distance is more then 5.

**Comment by eleurent** [comment]
Hi,
I think in this case the positions/velocities of other vehicles are **relative** to the ego vehicle (only the ego is absolute), which is the default, see 

https://github.com/Farama-Foundation/HighwayEnv/blob/edf4d17fa39a56ced4aabd209e45893382d4750e/highway_env/envs/common/observation.py#L148

You can change the observation config if you want absolute coordinates for all vehicles.

But that means that Vehicle2 is at relative position (-6.8, -1), which is essentially 1.8m behind the ego-vehicle. This is probably right after a collision occured.
You are not going to see the vehicles overlapping (dx < 5 and dy<2) from the coordinates in the observation anyway, since when this happens the vehicle bounce off each other.

(And on top of that is the policy frequency is 1Hz and simulation frequency is 15Hz (default), when you received the crashed signal in the info at the end of the step the collision might have happened earlier (at any frame since the last action, so up to 1s before), and the vehicles keep moving slightly after a crash.)

I hope that helps :)

**Comment by Davinho10** [comment]
Thank you this was very helpful

**Comment by Davinho10** [comment]
I have used the following notebook: https://colab.research.google.com/github/eleurent/highway-env/blob/master/scripts/sb3_highway_dqn.ipynb#scrollTo=Y5TOvonYqP-g
And as soon as I added the following configuration:config = {
"observation": {
"type": "Kinematics",
"features": ["presence", "x", "y", "vx", "vy"],
"normalize": False,
}
}
env.configure(config)
env.reset()
The videos where showing my agent crashing in every episode, after excluding "normalize": False, the agent performance was back to normal.
Do you have andy idea why this is the case?

**Comment by eleurent** [comment]
AFAIU it's just that neural networks tend to converge more easily when their inputs/outputs are roughly normalised in the [0, 1] range. You could probably keep the unnormalised inputs but add a LayerNorm / BatchNorm in the network architecture instead to achieve a similar effect, but I am not sure why you feel like disabling input normalisation altogether.
